core/domain/user_query_domain.py

```diff
@@ -21,22 +21,19 @@
 
 import collections
 
+from constants import constants
 import feconf
 import python_utils
 import utils
 
+__attribute_names__ = [
+    predicate['backend_attr'] for predicate in (
+        constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
 
-UserQueryParams = collections.namedtuple(
-    'UserQueryParams',
-    [
-        'inactive_in_last_n_days',
-        'has_not_logged_in_for_n_days',
-        'created_at_least_n_exps',
-        'created_fewer_than_n_exps',
-        'edited_at_least_n_exps',
-        'edited_fewer_than_n_exps'
-    ]
-)
+USER_QUERY_PARAMS = collections.namedtuple(
+    'USER_QUERY_PARAMS', __attribute_names__)
+USER_QUERY_PARAMS.__new__.__defaults__ = (None,) * len(
+    USER_QUERY_PARAMS._fields)
```

Comment:
Oh this is ugly, but seems like the correct solution. Can we add a TODO to change this after we migrate to Python 3?

Reply:
I have added the TODO. Done.